plugin/src/HeadlessAPI.lua

```diff
@@ -0,0 +1,452 @@
+local MarketplaceService = game:GetService("MarketplaceService")
+
+local Parent = script:FindFirstAncestor("Rojo")
+local Plugin = Parent.Plugin
+local Packages = Parent.Packages
+
+local Log = require(Packages.Log)
+
+local Config = require(Plugin.Config)
+local Settings = require(Plugin.Settings)
+local ApiContext = require(Plugin.ApiContext)
+
+local cloudIdInfoCache = {}
+local apiPermissionAllowlist = {
+	Version = true,
+	ProtocolVersion = true,
+	RequestAccess = true,
+}
+
+local API = {}
+
+function API.new(app)
+	local Rojo = {}
+
+	Rojo._rateLimit = {}
+	Rojo._sourceToPlugin = {}
+	Rojo._permissions = Settings:get("apiPermissions") or {}
+	Rojo._activePermissionRequests = {}
+	Rojo._changedEvent = Instance.new("BindableEvent")
+	Rojo._apiDescriptions = {}
+
+	Rojo._apiDescriptions.Changed = "An event that fires when a headless API property changes"
+	Rojo.Changed = Rojo._changedEvent.Event
+
+	Rojo._apiDescriptions.Connected = "Whether or not the plugin is connected to a Rojo server"
+	Rojo.Connected = if app.serveSession then app.serveSession:getStatus() == "Connected" else false
+
+	Rojo._apiDescriptions.Address = "The address (host:port) that the plugin is connected to"
+	Rojo.Address = nil
```

Comment:
I don't understand why `GetPortAndHost` exists when you can get the address here. Or does that return what the user has input whether they are connected or not? That could be more clear in the docs if that's the case.

Reply:
Address is the host and port that is currently connected and can be nil. GetHostAndPort returns whatever the plugin is set to, regardless of current connection. I will try to make this more clear, thanks for calling it out.